addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -523,15 +523,37 @@ export const editorCommands = {
     toggleList: (editor, mode) => {
         const li = new Set();
         const blocks = new Set();
+        const numberListRegex = /^\b(?:[1aA]\.\s*|[1aA]\))\s*/;
+        const bulletPointRegex = /^[\-*]+\s/;
 
         const selectedBlocks = getTraversedNodes(editor.editable);
         const deepestSelectedBlocks = selectedBlocks.filter(block => (
             !descendants(block).some(descendant => selectedBlocks.includes(descendant))
         ));
-        for (const node of deepestSelectedBlocks) {
+        for (let node of deepestSelectedBlocks) {
             if (node.nodeType === Node.TEXT_NODE && isWhitespace(node) && closestElement(node).isContentEditable) {
                 node.remove();
             } else {
+                // This code is placed here due to an issue encountered when it was written in the _onInput method.
+                // Although it worked correctly there, pressing CTRL + Z for the first time positioned my cursor at '[]1.'.
+                // To avoid this, the code has been relocated here.
+                let  closestBlockNode = closestBlock(node);
+
+                if (numberListRegex.test(closestBlockNode.textContent)) {
+                    // Use the closestBlock node as the node because the current node only contains '1.'.
+                    // For the space ' ', it is represented as a separate text node within the closestBlock node.
+                    node = closestBlockNode;
+                    node.textContent = node.textContent.replace(numberListRegex, '');
+                } else if (bulletPointRegex.test(closestBlockNode.textContent)) {
+                    // Use the closestBlock node as the node because the current node only contains '-' or '*'.
+                    // For the space ' ', it is represented as a separate text node within the closestBlock node.
+                    node = closestBlockNode;
+                    node.textContent = node.textContent.replace(bulletPointRegex, "");
+                }
+
+                // If the node's text content is empty, fill the closest element of node with br.
+                if (!node.textContent) fillEmpty(closestElement(node));
+
```

Comment:
No No, this code should be not a part of `toggleList`. About the issue you mentioned in comment did you tried using `_applyCommand` instead of `execCommand` which does not `computeHistorySelection`. Could you give it a try and check if it works?